FlowCrypt/Controllers/ImportKey/Enter Pass Phrase/EnterPassPhraseViewController.swift

```diff
@@ -200,25 +203,47 @@ extension EnterPassPhraseViewController {
     }
 
     private func checkIfKeyPossibleToAdd(with existedKeys: [KeyDetails], and passPhrase: String) {
-        let isKeyAlreadyAdded = Set(fetchedKeys).isSubset(of: existedKeys)
-
-        guard Set(fetchedKeys) != Set(existedKeys) else {
-            showAlert(message: "import_key_error_added_all".localized)
-            return
-        }
-
-        guard !isKeyAlreadyAdded else {
-            showAlert(message: "import_key_error_added".localized)
-            return
-        }
+        let keysToUpdate = Array(Set(existedKeys).intersection(fetchedKeys))
+        let newKeysToAdd = Array(Set(fetchedKeys).subtracting(existedKeys))
 
         keysDataService.addKeys(
-            keyDetails: fetchedKeys,
+            keyDetails: newKeysToAdd,
             passPhrase: passPhrase,
             source: .generated
```

Comment:
This is wrong - here the source of the key is imported, not generated

Reply:
Do I need to introduce a new case for it? 
Now we have 
```swift
enum KeySource {
    case backup
    case generated

    var path: String {
        switch self {
        case .backup: return "backup"
        case .generated: return "generated"
        }
    }
}
```

Reply:
I think that's a good idea. If the key was auto-loaded from backups, source should be "backup". If it was manually imported, then "imported" and if it was generated then "generated".